src/apps/loyalty/server/index.tsx

```diff
@@ -69,6 +70,18 @@ app.get("/inquiries", (req, res) => {
     return res.redirect(req.baseUrl + "/login")
   }
 
+  const headers = { "X-Access-Token": req.user.get("accessToken") }
```

Comment:
Oh yea for sure! This was more of a bare-bones example for others to provide context around the data that needs to be fetched.

I totally agree refactoring into a more suitable form is needed for this to be mergeable.

And yes, since we do need to fetch the collector profile (in addition to the user) for every route, moving it to a middleware as you suggest makes the most sense 👍 

I can do that!